qiskit/qasm2/__init__.py

```diff
@@ -0,0 +1,523 @@
+    ]
+    circuit = loads(program, custom_classical=customs)
+
+
+.. _qasm2-legacy-compatibility:
+
+Legacy Compatibility
+====================
+
+:meth:`.QuantumCircuit.from_qasm_str` and :meth:`~.QuantumCircuit.from_qasm_file` used to make a few
+additions on top of the raw specification. Qiskit originally tried to use OpenQASM 2 as a sort of
+serialisation format, and expanded its behaviour as Qiskit expanded.  The new parser under all its
+defaults implements the specification more strictly.
+
+The complete legacy code-paths are
+
+.. code-block:: python
+
+    from qiskit.converters import ast_to_dag, dag_to_circuit
+    from qiskit.qasm import Qasm
+
+    def from_qasm_file(path: str):
+        dag_to_circuit(ast_to_dag(Qasm(filename=path).parse()))
+
+    def from_qasm_str(qasm_str: str):
+        dag_to_circuit(ast_to_dag(Qasm(data=qasm_str).parse()))
+
+In particular, in the legacy importers:
+
+* the `include_path` is effectively:
+    1. ``<qiskit>/qasm/libs``, where ``<qiskit>`` is the root of the installed ``qiskit`` package;
+    2. the current working directory.
+
+* there are additional instructions defined in ``qelib1.inc``:
+    ``csx a, b``
+      Controlled :math:`\sqrt X` gate, corresponding to :class:`.CSXGate`.
+
+    ``cu(theta, phi, lambda, gamma) c, t``
+      The four-parameter version of a controlled-:math:`U`, corresponding to :class:`.CUGate`.
+
+    ``rxx(theta) a, b``
+      Two-qubit rotation arond the :math:`XX` axis, corresponding to :class:`.RXXGate`.
+
+    ``rzz(theta) a, b``
+      Two-qubit rotation arond the :math:`ZZ` axis, corresponding to :class:`.RZZGate`.
+
+    ``rccx a, b, c``
+      The double-controlled :math:`X` gate, but with relative phase differences over the standard
+      Toffoli gate.  This *should* correspond to the Qiskit gate :class:`~.RCCXGate`, but the legacy
+      converter wouldn't actually output this type.
+
+    ``rc3x a, b, c, d``
+      The triple-controlled :math:`X` gate, but with relative phase differences over the standard
+      definition.  Corresponds to :class:`.RC3XGate`.
+
+    ``c3x a, b, c, d``
+      The triple-controlled :math:`X` gate, corresponding to :class:`.C3XGate`.
+
+    ``c3sqrtx a, b, c, d``
+      The triple-controlled :math:`\sqrt X` gate, corresponding to :class:`.C3SXGate`.
+
+    ``c4x a, b, c, d, e``
+      The quadruple-controlled :math:`X` gate., corresponding to :class:`.C4XGate`.
+
+* if *any* ``opaque`` or ``gate`` definition was given for the name ``delay``, they attempt to
+  output a :class:`~qiskit.circuit.Delay` instruction at each call.  To function, this expects a
+  definition compatible with ``opaque delay(t) q;``, where the time ``t`` is given in units of
+  ``dt``.  The importer will raise errors on construction if there was not exactly one parameter
+  and one qubit, or if the parameter is not integer-valued.
+
+* the additional scientific-calculator functions ``asin``, ``acos`` and ``atan`` are available.
```

Comment:
```suggestion
* the trigonometric functions ``asin``, ``acos`` and ``atan`` are available.
```

Reply:
The point is that these extend the "scientific calculator" functionality (the paper's words).  The fact that they're all inverse trig functions is coincidental.